src/main/java/com/emc/pravega/perf/PravegaTransactionWriterWorker.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.emc.pravega.perf;
+
+import java.util.concurrent.CompletableFuture;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.stream.Transaction;
+import io.pravega.client.stream.TxnFailedException;
+
+public class PravegaTransactionWriterWorker extends PravegaWriterWorker {
+    private Transaction<String> transaction;
```

Comment:
`transaction` and `eventCount` below are accessed in the writeData method which is called from a forkJoin pool and hence may have call coming from multiple threads. But their assignments are not guarded by any lock, so this is a race condition. 
Specifically below, I can see this resulting in a potential NPE, a potential use of a transaction after commit, and multiple ways to miscount of events.

Reply:
ok, now evenCount is Atomic integer and transaction is the Atomic reference for the producer.begintransaction ; should be OK now.